typescript/infra/scripts/warp-routes/monitor/monitor-warp-routes-balances.ts

```diff
@@ -0,0 +1,239 @@
+import { PopulatedTransaction } from 'ethers';
+
+import {
+  ChainMap,
+  ChainMetadata,
+  CoinGeckoTokenPriceGetter,
+  EvmHypXERC20Adapter,
+  EvmHypXERC20LockboxAdapter,
+  IHypXERC20Adapter,
+  MultiProtocolProvider,
+  Token,
+  TokenStandard,
+  WarpCore,
+} from '@hyperlane-xyz/sdk';
+import { ProtocolType, objMerge } from '@hyperlane-xyz/utils';
+
+import { getWarpCoreConfig } from '../../../config/registry.js';
+import {
+  DeployEnvironment,
+  getRouterConfigsForAllVms,
+} from '../../../src/config/environment.js';
+import { fetchGCPSecret } from '../../../src/utils/gcloud.js';
+import { startMetricsServer } from '../../../src/utils/metrics.js';
+import { getArgs, withWarpRouteIdRequired } from '../../agent-utils.js';
+import { getEnvironmentConfig } from '../../core-utils.js';
+
+import {
+  metricsRegister,
+  updateTokenBalanceMetrics,
+  updateXERC20LimitsMetrics,
+} from './metrics.js';
+import { WarpRouteBalance, XERC20Limit } from './types.js';
+import { gracefullyHandleError, logger } from './utils.js';
+
+async function main() {
+  const { checkFrequency, environment, warpRouteId } =
+    await withWarpRouteIdRequired(getArgs())
+      .describe('checkFrequency', 'frequency to check balances in ms')
+      .demandOption('checkFrequency')
+      .alias('v', 'checkFrequency') // v as in Greek letter nu
+      .number('checkFrequency')
+      .parse();
+
+  startMetricsServer(metricsRegister);
+
+  const envConfig = getEnvironmentConfig(environment);
+  const registry = await envConfig.getRegistry();
```

Comment:
I think it could be useful to have an interactive step that warns us if the .registry commit is behind main on registry, we could enable ignoring the warning a proceeding anyway in special cases where want to use an old version of the registry.

I think it would be quite easy to forget to update the commit

Reply:
Because registry commits are so frequent I wonder if there's another heuristic we can go off of? Otherwise I'd expect this to just always complain

We could maybe have the deploy command make sure that the commit in the registryrc contains all the warp routes we want?

Reply:
The only downside to this is that the it may not pick up changes to configs, which would be very rare. If did happen that one time when we redeployed those eclipse routes. This may not be too much of a concern  

I think we would throw an error if we cant find the warp route id in registry anyway no?

Reply:
> I think we would throw an error if we cant find the warp route id in registry anyway no?

it would throw but wouldn't exit - fixed now!
I think this should help. Kinda inclined to kick the can on getting better registry integrations in this case, wdyt? We'd get alerted on this crashing and someone will fix it if there's an issue